### ISSUE TITLE
Incorrect X and Y touch points

### ISSUE THREAD

**Issue by prasad-dot-ws** [ploys/arduino-goodix#11]
Hi, Thanks for the awesome effort you put into this library. I tried to run this code. however, I'm getting very large numbers for X and Y positions. I have called the `readInfo()` function and it returned the correct resolution as 320 x 480.

This is the screen I'm using - http://www.lcdwiki.com/3.5inch_LCD_MODULE-Capacitive_Touch

This is what I'm getting when I touch the screen,

```cC0: #0 X = 65280 Y = 12032 s:0
C0: #0 X = 65280 Y = 12032 s:0
C0: #0 X = 65280 Y = 12032 s:0
C0: #0 X = 65280 Y = 12032 s:0
C0: #0 X = 65280 Y = 12032 s:0
C0: #0 X = 65280 Y = 12032 s:0
C0: #0 X = 65280 Y = 12032 s:0
C0: #0 X = 65280 Y = 12032 s:0
```
What could be the issue here?

**Comment by nomanshahsyed** [comment]
@prasad-dot-ws did you solve this issue already?

**Comment by u4mzu4** [comment]
@prasad-dot-ws , @nomanshahsyed 
I think I have a possible solution, which was tested on ESP32 and GT911 320x480.

**Comment by nomanshahsyed** [comment]
Yeah i managed to do a workaround too. Only needed to read the coordinates.

**Comment by Jakub-Nagy** [comment]
> I think I have a possible solution, which was tested on ESP32 and GT911 320x480.

Can you let me know what that is? I have the same setup: ESP32, GT911 and 320x480 screen.

> Yeah i managed to do a workaround too. Only needed to read the coordinates.

What do you mean by this? I worked with touchscreens before, and the values being big is not a problem, I can just map them.

The problem is my X and Y reading behave very weirdly. Look at what happens when I slide my finger across the X (480px long) axis:
![X axis](https://user-images.githubusercontent.com/37936430/114525558-7cc71e00-9c46-11eb-8ca0-626b4140bf8f.jpg)
The value goes up to 60000 and then, exactly in the middle of the screen, it goes back to zero and then again raises to around 50000. I was trying to move across the screen at a constant velocity.

The look at what happens with the Y reading when I move across the Y (320px long) axis:
![Y axis](https://user-images.githubusercontent.com/37936430/114525563-7df84b00-9c46-11eb-986c-159dade2213c.jpg)
The value just jumps at a couple of places. I actually found out that the X readings respond more linearly to touch on the Y axis, which I don't understand.

Can anyone help? Did you have similar/same patterns or were they different? Thanks.

**Comment by u4mzu4** [comment]
@Jakub-Nagy
Hello,
I've started to fix some issues and simplified the library. You can try this repo if you want:
https://github.com/u4mzu4/Arduino_GT911_Library

Focused on Lilygo Lilypi, but the example set it FALSE by default. Let me know, if you have any trouble with it.

**Comment by Jakub-Nagy** [comment]
Hi @u4mzu4 I've got the idea to try to look into your repos immediately after finding posting this issue. I've tried it out, it doesn't really work for me with the example.

I've tried just replacing the library with yours, the touch controller worked but with exactly the same issue.

I also tried the example sketch on your repo, this didn't work. I tried setting the Lilypi define to both true and false. Why is the reset pin undefined by default? Setting it up didn't help anyway. When I use your example with variations of these changes (Lilypi true/false and reset pin defined/undefined), it always fails on verifying the ACK on the addr request in the touchStart() method.

The touch.begin() method is returning true and the config is also always ok. I don't know where the issue is.

Can you maybe try to point out what the issue with the original library was? The interface, interrupts and detecting multitouch IDs work so I suppose there's just some config for the display missing. Thanks.

I'm not using the Lilypi but a very [similar display from Aliexpress](https://www.aliexpress.com/item/32814953314.html) (the ILI9488 CTP version).

**Comment by u4mzu4** [comment]
It's weird.
The original issue is the missing register processing, like this:
https://github.com/u4mzu4/Arduino_GT911_Library/blob/master/Goodix.cpp#L238-L261

Reset pin is not used in LilyPi, you can also try without this (connected directly), that's the reason why it's missing.
The I2C adress is 0x5D or 0x14?

Good luck, I hope you can find the rootcause.

**Comment by Jakub-Nagy** [comment]
> Good luck, I hope you can find the rootcause.

Thanks, I did, kind of.


> The I2C adress is 0x5D or 0x14?

I saw this and was like, oh yes, such a rookie mistake, because I remembered that the code is set to 0x5D but I remembered that I saw 0x14 on the I2C scanner. So I changed the address in the code, didn't work. I ran I2C scanner again, and to my surprise, the I2C address changed to 0x5D! I tested everything and I successfully got good X and Y readings. I thought that was the end.

I left my computer for a while and when I reconnected the dev board, it didn't work again and the I2C address went back to 0x14 and there were no touch interrupts. After some experimenting, I found that the original library can actually change the I2C address to 0x5D if you set it that way. The problem is the library from @u4mzu4 cannot do this. Meanwhile, the X/Y readings only work with the address 0x5D.

So, currently the only way I can make this work is to:
1} Load the code with the original library with the original example while connecting the reset pin and setting the address to 0x5D.
2) Change the library files for the updated library.
3) Disconnect the reset pin (not physically, only in pin mapping)
4) Load the new code


**UPDATE**
I fixed this problem by replacing the current reset() method with the original one, along with the pinSet(), pinOut(), pinIn() and pinHold() methods. I have no idea why this was the issue, but everything works perfectly for me know, even after reconnecting the power. Thanks for the help!

**Comment by u4mzu4** [comment]
Yes, in the GT911 you can set the I2C address with the help of the RESET pin. Since I've never used, I just simply removed, which was surely a mistake. I will put back for further usage.

Have you ever tried the example [line](https://github.com/u4mzu4/Arduino_GT911_Library/blob/master/examples/GT911_esp_touch/GT911_esp_touch.ino#L21)
with the address
GOODIX_I2C_ADDR_28  (instead of GOODIX_I2C_ADDR_BA)
Maybe no pin wobbling needed in this case.

**Comment by Jakub-Nagy** [comment]
> with the address
> GOODIX_I2C_ADDR_28 (instead of GOODIX_I2C_ADDR_BA)

Yes. But as mentioned earlier, without the original reset() function, this works only if the address was originally set to 0x5D (BA). If it was not, it fails. Funny thing is that even if I configured the address to 0x14, this didn't work. With the added original reset() function, both addresses work.

I'm attaching my modified library.
[arduino-goodix-master.zip](https://github.com/nik-sharky/arduino-goodix/files/6308533/arduino-goodix-master.zip)

**Comment by u4mzu4** [comment]
Magic, I have to check the datasheet again.
Anyway, great debugging, thanks for the support on behalf of the ones after you. I will migrate your solution as soon as I have some fee timeslot.
